source/code/plugin/health/health_monitor_state.rb

```diff
@@ -0,0 +1,204 @@
+module HealthModel
+
+    HealthMonitorInstanceState = Struct.new(:prev_sent_record_time, :old_state, :new_state, :state_change_time, :prev_records, :is_state_change_consistent, :should_send) do
+    end
+
+    # Class that is used to store the last sent state and latest monitors
+    # provides services like
+    # get_state -- returns the current state and details
+    # update_instance -- updates the state of the health monitor history records
+    # set_state -- sets the last health monitor state
+    class HealthMonitorState
+
+        def initialize
+            @@monitor_states = {}
+            @@first_record_sent = {}
+            @@health_signal_timeout = 240
```

Comment:
240 mins ?

Reply:
Yes. 4 hrs, which is the monitor timeout. Will be moved to a configuration later